### PR TITLE
(Ozone) Add scaling support based on 720p

### PR DESCRIPTION
**Pull request by frangarcj**
## Description

This PR adds scaling support to ozone considering 720p as 1.

Maybe I'm missing something to scale and it needs testing on other platforms.

Example on Vita. (544p)

## Reviewers
@natinusala

### CODE REVIEW

**Comment by natinusala** [comment]
This is great, thank you! I will try it and look at the code this week end

Does it scale upwards too? Like on a 4K monitor?

@twinaphex can you try this PR on your 4K monitor to see if it scales well?

Also, does it rescale when resizing the window like XMB does?

**Comment by frangarcj** [comment]
Code is based on XMB one so it should change when width changes

**Comment by jdgleaver** [comment]
Yes, this is lovely!

But I see you're using `settings->floats.menu_scale_factor` directly. It would be nice if you could try implementing this preferably with `menu_display_get_dpi_scale()`, or `menu_display_get_pixel_scale()` if DPI-based scaling doesn't look right.

**Comment by twinaphex** [comment]
@frangarcj Is it possible for you to look into what @jdgleaver is suggesting here?

**Comment by frangarcj** [comment]
I'll add the changes

**Comment by jdgleaver** [comment]
@frangarcj Thanks for adding the DPI-based scaling!

Unfortunately, I've just tested the PR, and it's not quite there yet...

One quick omission is that you can't set the menu scale factor - this is fixed with the following diff:

```diff
diff --git a/menu/menu_setting.c b/menu/menu_setting.c
index 2af5c1d05a..6cc647780b 100644
--- a/menu/menu_setting.c
+++ b/menu/menu_setting.c
@@ -12423,9 +12423,10 @@ static bool setting_append_list(
 
          START_SUB_GROUP(list, list_info, "Display", &group_info, &subgroup_info, parent_group);
 
-         /* Only implemented for GLUI and XMB at present */
+         /* Only implemented for GLUI, XMB and Ozone at present */
          if (string_is_equal(settings->arrays.menu_driver, "glui") ||
-             string_is_equal(settings->arrays.menu_driver, "xmb"))
+             string_is_equal(settings->arrays.menu_driver, "xmb") ||
+             string_is_equal(settings->arrays.menu_driver, "ozone"))
             CONFIG_FLOAT(
                   list, list_info,
                   &settings->floats.menu_scale_factor,
```

...but more importantly, there are several things that just don't scale correctly. For example:

- Left and right side bar widths are wrong - this is very obvious when the scale is large or the window is small

- Word wrap width (for sublabels) is wrong - more specifically, if you change the window size or scale factor at runtime, sublabels break completely

- If you change window size or scale factor and then navigate backwards, the resultant menu list is messed up

- This is what the menu looks like by default on a 15" 768p display:

Screenshot_2019-11-16_14-19-55

That's probably not intended behaviour!

Basically, this is a fantastic start, but I think it just needs going though a little more rigorously, to double-check every dimension parameter. I know this is annoying stuff, but the end result will be awesome!

**Comment by frangarcj** [comment]
Thanks. Little by little we'll get it to be ok

**Comment by twinaphex** [comment]
Can now be comfortably closed with @jdgleaver's excellent PR.

**Comment by natinusala** [comment]
Does it downscale? I've had multiple Vita users ask me if the menu would be usable now

**Comment by jdgleaver** [comment]
@natinusala Yes, absolutely! Ozone should scale fine in both directions now
